pkg/reconciler/artifactbuildrequest/artifactbuildrequest.go

```diff
@@ -59,43 +67,91 @@ func (r *ReconcileArtifactBuildRequest) Reconcile(ctx context.Context, request r
 	tr.Namespace = abr.Namespace
 	tr.GenerateName = abr.Name + "-scm-discovery-"
-	tr.Labels = map[string]string{TaskRunLabel: ""}
+	tr.Labels = map[string]string{ArtifactBuildRequestIdLabel: string(abr.UID), TaskRunLabel: ""}
 	tr.Spec.Params = append(tr.Spec.Params, pipelinev1beta1.Param{Name: "GAV", Value: pipelinev1beta1.ArrayOrString{Type: pipelinev1beta1.ParamTypeString, StringVal: abr.Spec.GAV}})
-	if err := controllerutil.SetOwnerReference(&abr, &tr, r.scheme); err != nil {
+	if err := controllerutil.SetOwnerReference(abr, &tr, r.scheme); err != nil {
 		return reconcile.Result{}, err
 	}
 	if err := r.client.Create(ctx, &tr); err != nil {
 		return reconcile.Result{}, err
 	}
 	abr.Status.State = v1alpha1.ArtifactBuildRequestStateDiscovering
-	if err := r.client.Status().Update(ctx, &abr); err != nil {
+	if err := r.client.Status().Update(ctx, abr); err != nil {
 		return reconcile.Result{}, err
 	}
 	return reconcile.Result{}, nil
 }
 
-func (r *ReconcileArtifactBuildRequest) handleStateDiscovered(ctx context.Context, abr v1alpha1.ArtifactBuildRequest) (reconcile.Result, error) {
+func (r *ReconcileArtifactBuildRequest) handleStateDiscovering(ctx context.Context, abr *v1alpha1.ArtifactBuildRequest) (reconcile.Result, error) {
+	log := log2.FromContext(ctx)
+	log.Info("Discovering: ", "abr", abr.Name)
+	//lets look up our discovery task
+	listOpts := &client.ListOptions{
+		Namespace:     abr.Namespace,
+		LabelSelector: labels.SelectorFromSet(map[string]string{ArtifactBuildRequestIdLabel: string(abr.UID)}),
+	}
+	trl := pipelinev1beta1.TaskRunList{}
+	err := r.client.List(ctx, &trl, listOpts)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	var tr *pipelinev1beta1.TaskRun
+	//there should only be one, be guard against multiple
+	for _, current := range trl.Items {
+		if tr == nil || tr.CreationTimestamp.Before(&current.CreationTimestamp) {
+			tr = &current
+		}
+	}
+	if tr == nil {
+		//no pipeline, very odd
+		//move back to new
+		abr.Status.State = v1alpha1.ArtifactBuildRequestStateNew
+		return reconcile.Result{}, r.client.Status().Update(ctx, abr)
+	}
+	if tr.Status.CompletionTime == nil {
+		return reconcile.Result{}, nil
+	}
+
+	//we grab the results here and put them on the ABR
+	for _, res := range tr.Status.TaskRunResults {
+		switch res.Name {
+		case TaskResultScmUrl:
+			abr.Status.SCMURL = res.Value
+		case TaskResultScmTag:
+			abr.Status.Tag = res.Value
+		case TaskResultScmType:
+			abr.Status.SCMType = res.Value
+		case TaskResultMessage:
+			abr.Status.Message = res.Value
+		case TaskResultContextPath:
+			abr.Status.Path = res.Value
+		}
+	}
+
 	//now let's create the dependency build object
 	//once this object has been created its resolver takes over
 	if abr.Status.Tag == "" {
 		//this is a failure
 		abr.Status.State = v1alpha1.ArtifactBuildRequestStateMissing
-		return reconcile.Result{}, r.client.Status().Update(ctx, &abr)
+		return reconcile.Result{}, r.client.Status().Update(ctx, abr)
```

Comment:
can you put a TODO comment to remind me to generate an event when this happens

I'll handle dealing with events in a separate PR